app/main.py

```diff
@@ -3,42 +3,60 @@
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
+import asyncio
 
+from typing import Any, Awaitable
 
-def main() -> None:
+
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> None:
+    await asyncio.gather(*functions)
+
+
+async def main() -> None:
     # create an IOT service
     service = IOTService()
 
-    # create and register a few devices
+    # run_parallel
     hue_light = HueLightDevice()
     speaker = SmartSpeakerDevice()
     toilet = SmartToiletDevice()
-    hue_light_id = service.register_device(hue_light)
-    speaker_id = service.register_device(speaker)
-    toilet_id = service.register_device(toilet)
-
+    hue_light_id = asyncio.create_task(service.register_device(hue_light))
+    speaker_id = asyncio.create_task(service.register_device(speaker))
+    toilet_id = asyncio.create_task(service.register_device(toilet))
```

Comment:
```suggestion
``` hue_light_id, speaker_id, toilet_id = await asyncio.gather(
        service.register_device(hue_light),
        service.register_device(speaker),
        service.register_device(toilet)
    )
```

Reply:
thank you